### ISSUE TITLE
[styled-components] support forwardedAs prop

### ISSUE THREAD

**Issue by deini** [DefinitelyTyped/DefinitelyTyped#40229]
- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @Igorbek @Igmat @lavoaster @Jessidhia @jkillian @wagerfield @eps1lon @flavordaaave

`forwardedAs` prop definition https://github.com/styled-components/styled-components/pull/2580

**Comment by xumix** [comment]
Have problems with ` @types/styled-components@4.1.22` and `semantic-ui-react@0.88.1`

```
import styled from "styled-components/macro";
import { Header } from "semantic-ui-react";
export const RegistrationLink = styled(Header)`
  color: #000 !important;
  font-weight: 400 !important;
  display: block;
  cursor: pointer;
`;
```
```
<RegistrationLink as="a" textAlign="center" size="tiny" onClick={this.goToRegister} />
```
Getting this error:
```
No overload matches this call.
  Overload 1 of 2, '(props: Pick<Pick<PropsWithChildren<HeaderProps>, ReactText> & Partial<Pick<PropsWithChildren<HeaderProps>, never>>, ReactText> & { ...; } & { ...; }): ReactElement<...>', gave the following error.
    Type 'string' is not assignable to type 'undefined'.
  Overload 2 of 2, '(props: StyledComponentPropsWithAs<"a", any, {}, never>): ReactElement<StyledComponentPropsWithAs<"a", any, {}, never>, string | ... 1 more ... | (new (props: any) => Component<...>)>', gave the following error.
    Type '{ children: any; as: "a"; textAlign: string; size: string; onClick: () => void; }' is not assignable to type 'IntrinsicAttributes & Pick<Pick<Pick<DetailedHTMLProps<AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement>, "children" | ... 261 more ... | "referrerPolicy"> & { ...; }, "children" | ... 262 more ... | "referrerPolicy"> & Partial<...>, "children" | ... 262 more ... | "referrerPolicy"> & { ...; } & { ...; }'.
      Property 'textAlign' does not exist on type 'IntrinsicAttributes & Pick<Pick<Pick<DetailedHTMLProps<AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement>, "children" | ... 261 more ... | "referrerPolicy"> & { ...; }, "children" | ... 262 more ... | "referrerPolicy"> & Partial<...>, "children" | ... 262 more ... | "referrerPolicy"> & { ...; } & { ...; }'.  TS2769
```

**Comment by xumix** [comment]
Forgot to mention authors: @Igorbek @Igmat @Lavoaster @Jessidhia @JKillian @wagerfield @eps1lon @flavordaaave

**Comment by deini** [comment]
@xumix your issue is not the same as the reported issue in the description. Try opening a different issue.

**Comment by bodinsamuel** [comment]
Seems like it was fixed by #41347

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
